src/services/user.js

```diff
@@ -1,19 +1,92 @@
 import User from '../models/user';
-
+import ExceptionUtils from '../utils/exception';
 class UserService {
 	constructor() {
 		this.userModel = User.getInstance();
 		this.userReadModel = User.getInstance('replica');
 	}
 
-	list() {
-		const users = this.userReadModel.findAll({
+	async list(data) {
+		if (data) {
+			const user = await this.userReadModel.findOne({
+				where: {
+					id: data,
+					is_deleted: false
+				}
+			});
+
+			if (!user) {
+				throw new ExceptionUtils(400, 'USER_NOT_FOUND');
+			}
+
+			return user;
+		}
+
+		return this.userReadModel.findAll({
+			where: {
+				is_deleted: false
+			}
+		});
+	}
+
+	async create(data) {
+		const user = await this.userReadModel.findOne({
 			where: {
+				name: data.name,
 				is_deleted: false
 			}
 		});
 
-		return users;
+		if (user) {
+			throw new ExceptionUtils(400, 'NAME_IN_USE');
```

Comment:
Pode ser interessante colocar uma mensagem como:

```suggestion
			throw new ExceptionUtils(400, 'Invalid user');
```

Desse modo, não enviaremos nada para o client que dê a entender que exista algum usuário

Reply:
Acho que a mensagem pode ser em português também